cli/internal/cmd/upgradeapply.go

```diff
@@ -303,6 +303,7 @@ func parseTerraformUpgradeVars(cmd *cobra.Command, conf *config.Config, fetcher
 			Debug:   conf.IsDebugCluster(),
 		}
 		return targets, vars, nil
+	// TODO support for QEMU?
```

Comment:
To support upgrades on QEMU we would need to implement a lot of custom logic and solutions.
Especially things that run "outside" of the Constellation context, i.e. not a service that runs in the Kubernetes cluster, or as part of the CLI, but some service that runs on the host and interacts with QEMU/libvirt to scale and update resources.

This is something we probably don't want to do, since the effort for implementing and keeping it functional would be a lot higher than the use we get out of it.
QEMU clusters should be used for testing only, for more integrated on-prem solutions,  we support Constellation on OpenStack (although I think OpenStack also does not support upgrades at the current time)

Reply:
thx for clarifying